### PR TITLE
Feat: Updated the Nav merged the both profile and nav into one nav fo…

### LINKED ISSUE #1301: [Feature]: Need only one menu on Navbar
### What happened?

We currently have two menu(s) on the nav (for mobile) 
1. For Profile
Screenshot from 2023-12-10 14-11-03
3. For Nav Links
image

We need only one menu, having two menus can lead to confusion and is a waste of space.
 
Expected:
1. Put the three links (from profile menu) at the bottom on nav menu.
2. Remove the icons, keep the red color for logout.

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PraveenShinde3**
feat(navbar): Combine profile and navigation menus for mobile view

## Related Issue
Closes  #1301 

## Motivation and Context
Addresses the issue of having only one menu on the navbar. Merged the profile menu and navigation menu to create a unified and mobile-friendly navigation experience.

## How Has This Been Tested?
Tested on various devices and browsers to ensure consistent behavior and responsiveness.

## Screenshots/Video (if applicable):

For User
image

For Admin
image

### CODE REVIEW

**Comment by odicho** [comment]
Nice! I think it still feels like something is missing. Are you able to experiment a little with the spacing / see if having a header for each section looks nice? Otherwise, as a first iteration, I think this is definitely an improvement!!

cc: @PickleNik @bautistaaa

**Comment by bautistaaa** [comment]
this is probably good enough for a first iteration

**Comment by bautistaaa** [comment]
you need to run `pnpm checks:fix`

File: apps/web/src/components/Navigation/nav-link.tsx [hunk 1]
```diff
@@ -3,10 +3,18 @@ import clsx from 'clsx';
...
+  mobileView,
+}: {
+  href: string;
+  title: string;
+  mobileView?: boolean;
+}) {
   const pathname = usePathname();
   return (
-    <Link href={href} className="ml-4">
+    <Link href={href} className={mobileView ? 'ml-4 md:hidden' : 'ml-4'}>
```
Comment (aadamw) [inline apps/web/src/components/Navigation/nav-link.tsx:17]: q: Why this can't be handled with just CSS?

File: apps/web/src/components/Navigation/index.tsx [hunk 2]
```diff
@@ -19,6 +19,10 @@ import { MobileNav } from './mobile-nav';
 import { NavLink } from './nav-link';
 import { Badge } from '@repo/ui/components/badge';
 
+const session = await auth();
```
Comment (bautistaaa) [inline apps/web/src/components/Navigation/index.tsx:22]: this should be in the context of the component and not global

File: apps/web/src/components/Navigation/index.tsx [hunk 3]
```diff
@@ -19,6 +19,10 @@ import { MobileNav } from './mobile-nav';
...
+
+const isAdminOrMod = isAdminOrModerator(session);
```
Comment (bautistaaa) [inline apps/web/src/components/Navigation/index.tsx:24]: move inside a component

**Review by bautistaaa — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Navigation/index.tsx [hunk 4]
```diff
@@ -38,13 +42,32 @@ export async function Navigation() {
...
       {featureFlags?.enableHolidayEvent ? (
-        <div className="flex items-center gap-1">
+        <div className="flex items-center gap-1 ">
           <NavLink title={t('advent')} href="/aot-2023" />
           <Badge className="h-4 bg-red-600 px-1.5" variant="default">
             New
           </Badge>
         </div>
       ) : null}
+      {session && session.user && (
```
Comment (MichaelG824) [inline apps/web/src/components/Navigation/index.tsx:52]: You can shorten this to use optional chaining e.g session?.user instead.

**Comment by bautistaaa** [comment]
did you plan on finishing this

**Comment by bautistaaa** [comment]
made it good enough ™

**Review by bautistaaa — APPROVED** [review]
